addons/website_payment/controllers/portal.py

```diff
@@ -65,14 +65,24 @@ def donation_transaction(self, amount, currency_id, partner_id, access_token, mi
             amount=amount, currency_id=currency_id, partner_id=partner_id, **kwargs
         )
         tx_sudo.is_donation = True
-        if use_public_partner:
-            tx_sudo.update({
-                'partner_name': details['name'],
-                'partner_email': details['email'],
-                'partner_country_id': int(details['country_id']),
-            })
-        elif not tx_sudo.partner_country_id:
-            tx_sudo.partner_country_id = int(kwargs['partner_details']['country_id'])
+
+        fields = tx_sudo._fields
+        extra_fields = {}
+        for key, value in details.items():
+            # Find matching field key
+            if key in fields:
+                # Type conversion for specific fields
+                field_type = fields[key].type
+                if field_type in ('many2one'):
+                    value = int(value) if value else False
+                tx_sudo.update({key: value})
+            else:
+                extra_fields[key] = value
+        if extra_fields:
+            tx_sudo.extra_fields = extra_fields
+
+        if not use_public_partner and not tx_sudo.partner_country_id:
+            tx_sudo.partner_country_id = int(kwargs.get('partner_details', {}).get('partner_country_id'))
```

Comment:
it s not possible to add the message in the chatter here ? If you can do that, we don t need to add the field `extra_fields`